apps/api/views.py

```diff
@@ -455,14 +467,14 @@ def trigger_bot_message(request):
 
     experiment = get_object_or_404(Experiment, public_id=experiment_public_id, team=request.team)
 
-    try:
-        participant_data = ParticipantData.objects.defer("data").get(
-            participant__identifier=identifier,
-            participant__platform=platform,
-            experiment=experiment.id,
-        )
-
-    except ParticipantData.DoesNotExist:
+    participant_data = ParticipantData.objects.filter(
+        participant__identifier=identifier,
+        participant__platform=platform,
+        experiment=experiment.id,
+    ).first()
+    if platform == ChannelPlatform.COMMCARE_CONNECT and not participant_data:
+        return Response({"detail": "Participant not found"}, status=status.HTTP_404_NOT_FOUND)
+    elif not Participant.objects.filter(identifier=identifier, platform=platform).exists():
```

Comment:
Can you give more context on this change - it doesn't seem related to the rest of the PR.

Reply:
This endpoint will not work for non-connect channels unless they have particpant data, which is typically the case for non-connect channels.

Reply:
I think one or two is great but this one doesn't add much value over the basic one. It just shows the one additional field.

Reply:
A comment in code would be useful - is it because connect is a 'push' channel? Maybe we can make that more explicit.

Reply:
The main reason that connect requires participant data is because we save the consent, channel id and encryption key on there. I'll add a comment